pkg/authz/basic_auth_authorizer.go

```diff
@@ -0,0 +1,142 @@
+package authz
+
+import (
+	"crypto/subtle"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	"github.com/pkg/errors"
+)
+
+type basicAuthAuthorizer struct {
+	nodeID string
+	// Separate maps for different authentication methods
+	basicAuthUsers map[string]types.AuthUser // Key is username
+	// Capability checker for authorization
+	capabilityChecker *CapabilityChecker
+	// Endpoint permissions mapping
+	endpointPermissions map[string]string
+}
+
+// validateBasicAuth validates basic authentication credentials
+func (a *basicAuthAuthorizer) validateBasicAuth(authHeader string) (types.AuthUser, bool, error) {
+	// Extract and decode the credentials
+	encodedCredentials := authHeader[6:] // Skip "Basic "
+	decodedBytes, err := base64.StdEncoding.DecodeString(encodedCredentials)
+	if err != nil {
+		return types.AuthUser{}, false, errors.Wrap(err, "failed to decode basic auth credentials")
+	}
+
+	// Split the credentials into username and password
+	credentials := string(decodedBytes)
+	parts := strings.SplitN(credentials, ":", 2)
+	if len(parts) != 2 {
+		return types.AuthUser{}, false, errors.New("invalid basic auth credentials format")
+	}
+
+	username, password := parts[0], parts[1]
+
+	// Look up the user by username (case-insensitive)
+	user, exists := a.basicAuthUsers[strings.ToLower(username)]
+	if !exists {
+		return types.AuthUser{}, false, errors.New("invalid basic auth credentials")
+	}
+
+	// Compare passwords using constant-time comparison to prevent timing attacks
+	if subtle.ConstantTimeCompare([]byte(user.Password), []byte(password)) != 1 {
+		return types.AuthUser{}, false, errors.New("invalid basic auth credentials")
+	}
+
+	// Authentication successful
+	return user, true, nil
+}
+
+func NewBasicAuthAuthorizer(
+	nodeID string,
```

Comment:
why do we need to the nodeID in auth?

Reply:
This is for future node specific JWT tokens (still thinking about the best way to add it)